src/System.Text.Json/src/System/Text/Json/Serialization/MemberAccessor.cs

```diff
@@ -3,91 +3,13 @@
 // See the LICENSE file in the project root for more information.
 
 using System.Reflection;
-using System.Runtime.CompilerServices;
 
 namespace System.Text.Json.Serialization
 {
-    internal static class MemberAccessor
+    internal abstract class MemberAccessor
```

Comment:
Wouldn't this indirection via an abstract class cause some perf regression?

Reply:
No, it affects class info construction only. A constructed class info is cached in the associated `JsonSerializerOptions` and reused later. The class materializer is implemented the same way, so I'm going to merge them.

Speaking about performance, there should be an improvement because the inbox build uses reflection emit to generate stubs to unbox an object instance and to pass it to the actual getter or setter. This eliminates one delegate call per action.